client/lib/signup/step-actions.js

```diff
@@ -253,27 +253,49 @@ module.exports = {
 			} );
 		} else {
 			wpcom.undocumented().usersNew( assign(
-				{}, userData, {
-					ab_test_variations: getSavedVariations(),
-					validate: false,
-					signup_flow_name: flowName,
-					nux_q_site_type: surveySiteType,
-					nux_q_question_primary: surveyVertical,
-					jetpack_redirect: queryArgs.jetpackRedirect
-				}
-			), ( error, response ) => {
-				const errors = error && error.error ? [ { error: error.error, message: error.message } ] : undefined,
-					bearerToken = error && error.error ? {} : { bearer_token: response.bearer_token };
-
-				if ( ! errors ) {
-					// Fire after a new user registers.
-					analytics.tracks.recordEvent( 'calypso_user_registration_complete' );
-					analytics.ga.recordEvent( 'Signup', 'calypso_user_registration_complete' );
-				}
+			{}, userData, {
+				ab_test_variations: getSavedVariations(),
+				validate: false,
+				signup_flow_name: flowName,
+				nux_q_site_type: surveySiteType,
+				nux_q_question_primary: surveyVertical,
+				jetpack_redirect: queryArgs.jetpackRedirect
+			}
+		), ( error, response ) => {
+			var errors = error && error.error ? [ { error: error.error, message: error.message } ] : undefined,
+				bearerToken = error && error.error ? {} : { bearer_token: response.bearer_token };
+
+			if ( ! errors ) {
+				// Fire after a new user registers.
+				analytics.tracks.recordEvent( 'calypso_user_registration_complete' );
+				analytics.ga.recordEvent( 'Signup', 'calypso_user_registration_complete' );
+				/**
+				 * Auto login the user when it has been created.
+				 */
+				user.clear();
+				user.fetching = false;
+
+				user.on( 'change', () => {
+					const newLoggedUser = user.get();
+
+					if ( newLoggedUser && newLoggedUser.ID ) {
+						analytics.identifyUser( newLoggedUser.ID, newLoggedUser.username );
+					} else {
+						/**
+						 * The user fetching is a bit fragile and sometimes it requires to do a double-fetch.
+						 */
+						user.clear();
+						user.fetching = false;
+						user.fetch();
+					}
+				} );
 
-				callback( errors, assign( {}, { username: userData.username }, bearerToken ) );
-			} );
-		}
+				// Force fetch the user details
+				user.fetch();
+			}
+
+			callback( errors, assign( {}, { username: userData.username }, bearerToken ) );
+		} );
```

Comment:
It looks like we lost an indentation level at L256, and thus lost the closing bracket for the conditional on L54.